### PR TITLE
Table: add select and selectAll checkboxes

### PR DESCRIPTION
**Pull request by Whapow**
# Link to Github Issue

Addresses: https://github.com/AmpleOrganics/Blaze.vue/issues/324

# Description

Adds support for using ao-table to select options from a list.

### CODE REVIEW

File: src/components/AoTable.vue [hunk 1]
```diff
@@ -2,6 +2,14 @@
   <table :class="[computedClasses, computedVAlign]">
     <thead>
       <tr>
+        <th v-if="selectionTable">
```
Comment (sarngru) [inline src/components/AoTable.vue]: name suggestion: `selectableTable`

File: src/components/AoTable.vue [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+          <input
+            v-model="selectAll"
+            type="checkbox"
+            :indeterminate.prop="partialSelect"
+            @change="$emit('massSelect', selectAll)"
```
Comment (sarngru) [inline src/components/AoTable.vue]: event name suggestion `selectAll` instead of `massSelect`

File: docs/components/Table.vue [hunk 3]
```diff
@@ -13,13 +13,23 @@
           :no-data-text="noDataText"
           :sort-by="sortBy"
           :order="order"
+          :selection-table="selectionTable"
+          :partial-select="partialSelect"
```
Comment (sarngru) [inline docs/components/Table.vue]: name suggestion: `is-partial-select`

File: docs/components/Table.vue [hunk 4]
```diff
@@ -13,13 +13,23 @@
...
           class="component-example-table"
           @sortTable="sortTable"
+          @massSelect="massSelection($event)"
```
Comment (sarngru) [inline docs/components/Table.vue]: `all` seems more suitable than `mass`

**Comment by ZainW** [comment]
@busheikin can you take a look at this and see if it works well for the meantime for something in your designs?

File: docs/components/Table.vue [hunk 5]
```diff
@@ -13,13 +13,23 @@
...
         >
           <tr
             v-for="user in filteredUsers"
             :key="user.id"
+            @click="toggleSelection(user)"
```
Comment (busheikin) [inline docs/components/Table.vue]: I think we should take this out even though it's just a demo, I don't want to suggest that clicking a row to select is our recommended usage

**Comment by busheikin** [comment]
It would be great to have a class added to a row if selected so it could be styled. Not sure how that could factor in to this approach. It might be wise to make a table-row component, and it could be given a selectable prop, and get a little more hands on with the usage.

**Comment by ZainW** [comment]
will approve once conflicts are fixed @Whapow

**Pull request merged**